### ISSUE TITLE
OIDC SSO automatic user creation

### ISSUE THREAD

**Issue by JoTec2002** [knadh/listmonk#2119]
**Is your feature request related to a problem? Please describe.**
I have sucessfully integrated in version v4.0.1 SSO and I'm very greatful for this feature.
But I have a large and changing number of users in my SSO Server and it is a bit inconvenient to create every user again in listmonk to enable SSO for this user.

**Describe the solution you'd like**
It would be very helpful to have the option to automaticly create a new user with a default role.
So like a checkbox under Settings > Security > OIDC "create new user from SSO if not existend"
and an dropdown to than assign a default role to every new automaticly created SSO user

**Comment by gramakri** [comment]
We tried to integrate this feature into Cloudron's package. However, the lack of automatic user creation makes it difficult.

How is one supposed to use OIDC without automatic user creation? The admin should create users manually first?

**Comment by knadh** [comment]
For now, yes, the admin has to create the users and OIDC is connected via email. Automatic user creation + default role is planned for the the next version.

A newsletter manager is highly likely to have a very small, closed group of users with specific, tailored permissions. Does your usecase involve a sizeable number of users?

**Comment by gramakri** [comment]
@knadh fair question . Personally, for our set up, we don't have a use case of many users. We can add them manually.

With my Cloudron packager hat on, I have no idea how the listmonk package is used :) We don't track user count or user related info in the platform. But based on your comment, we decided to go ahead and release an initial package with OIDC support and asking the user to add listmonk oidc users manually. We can also confirm that v4 oidc works with Cloudron OIDC . Thanks!

**Comment by centja1** [comment]
@knadh If you do decide to implement automatic user creation, might I also suggest implementing configurable scopes on OIDC login

Listmonk clearly has the concept of roles (Super-Admin, etc) and it would be great from our end to be able to manage the grants in our IdP and have Listmonk read them from a claim in the ID token.

For example, our standard pattern for OIDC connected applications is to create a custom scope with the application name, i.e. "listmonk".  Then, the application can request that scope for a given user and it has the claim in the structure needed by the application to read roles, etc.

From looking at the code, it appears listmonk is storing User/Role associations in the database, but the beauty of custom scopes is that the IdP can return the custom claim in any format you need, so you could theoretically put constraints on the format of the custom claim to reduce coding effort.

I'd be happy to help test out your implementations if you decide to do this.  Thanks for the great project!

**Comment by gramakri** [comment]
Not stale

**Comment by ski7777** [comment]
Is there any plan to implement this?

**Comment by dacrudg** [comment]
Yes, this would really be useful.  Right now the SSO implementation is a pain to have to go in and create every user.
Thank you

**Comment by violoncelloCH** [comment]
Hi @knadh 
Could this be reopened ? It would be really useful to allow for automatic user provisioning from oidc and also if roles could be defined through the oidc grant

**Comment by knadh** [comment]
Yep, this is planned for the next version.

**Comment by knadh** [comment]
This is merged in #2578.

**Comment by ski7777** [comment]
Thanks you :)

Kailash Nadh ***@***.***> schrieb am So., 20. Juli 2025,
12:42:

**Comment by violoncelloCH** [comment]
Thank you very much @vividvilla and @knadh for implementing this!

I've created #2589 for the other point that was mentioned here: assigning the roles through OIDC
